### PR TITLE
Publish umd and minified umd versions of package

### LINKED ISSUE #35: Precompiled version?
I'm trying to use this on a site that doesn't use Node/NPM; is there any chance you could put up a pre-compiled version for these types of use cases?

### PR DESCRIPTION
**Pull request by vladimyr**
Implementing browserify based build system for producing (minified)
umd builds suitable for unpkg.com. (resolves #35)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,13 +2,18 @@
   "name": "focus-trap",
   "version": "2.4.3",
   "description": "Trap focus within a DOM node.",
-  "main": "index.js",
+  "main": "dist/focustrap.min.js",
```
Comment (davidtheclark) [inline package.json]: I'd rather leave the unminified version as the file that people get if they `require` the package. Do you think there's a reason to prefer the minified version?

File: package.json [hunk 2]
```diff
@@ -2,13 +2,18 @@
...
     "lint": "eslint .",
     "demo-bundle": "browserify demo/js/index.js -o demo/demo-bundle.js",
+    "clean": "del-cli !dist/.gitkeep dist/*",
+    "build-dev": "browserify index.js -s focusTrap > dist/focustrap.js",
+    "minify": "uglifyjs dist/focustrap.js > dist/focustrap.min.js",
+    "build": "npm run build-dev && npm run minify",
     "start": "budo demo/js/index.js:demo-bundle.js --dir demo --live",
-    "test": "npm run lint"
+    "test": "npm run lint",
+    "prepare": "npm run build"
```
Comment (davidtheclark) [inline package.json]: How about `prepublishOnly`?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,5 @@
 node_modules
 *.log
 **/*-bundle.js
+dist/*
+!dist/.gitkeep
```
Comment (davidtheclark) [inline .gitignore]: Won't these new lines cause the built files to be ignored by npm, so they won't show up in unpkg.com?

File: package.json (hunk 1 above)
Comment (vladimyr) [reply package.json]: Actually you are right, I'll twist it.

File: package.json (hunk 2 above)
Comment (vladimyr) [reply package.json]: Yeah, I took wrong turn here, I'll change ti to `prepublishOnly`. :+1:

File: .gitignore (hunk 3 above)
Comment (vladimyr) [reply .gitignore]: I forgot that `npm` actually respects `.gitignore` if no `.npmignore` is found and/or `package.json`'s `files` property is not defined. Bottom line, you are right but I'm not sure whether to whitelist stuff using `files` or blacklist using `.npmignore`. Any preferences? :thinking:

Comment (vladimyr) [reply .gitignore]: _Answering to myself:_ You already used `files` so I just extended it...

**Comment by vladimyr** [comment]
FYI This is what I get after running:
`$ npm run prepublishOnly && npm pack`

```
> focus-trap@2.4.3 prepublishOnly /home/vladimyr/Projects/_private/focus-trap
> npm run build


> focus-trap@2.4.3 build /home/vladimyr/Projects/_private/focus-trap
> npm run build-dev && npm run minify


> focus-trap@2.4.3 build-dev /home/vladimyr/Projects/_private/focus-trap
> browserify index.js -s focusTrap > dist/focustrap.js


> focus-trap@2.4.3 minify /home/vladimyr/Projects/_private/focus-trap
> uglifyjs dist/focustrap.js > dist/focustrap.min.js

npm notice 
npm notice 📦  focus-trap@2.4.3
npm notice === Tarball Contents === 
npm notice 1.3kB  package.json         
npm notice 1.4kB  CHANGELOG.md         
npm notice 2.7kB  index.d.ts           
npm notice 6.7kB  index.js             
npm notice 1.1kB  LICENSE              
npm notice 7.3kB  README.md            
npm notice 0      dist/.gitkeep        
npm notice 10.1kB dist/focustrap.js    
npm notice 7.5kB  dist/focustrap.min.js
npm notice === Tarball Details === 
npm notice name:          focus-trap                              
npm notice version:       2.4.3                                   
npm notice filename:      focus-trap-2.4.3.tgz                    
npm notice package size:  9.1 kB                                  
npm notice unpacked size: 38.2 kB                                 
npm notice shasum:        2c2c012d030a4b2a261eacf5fb5977d0613f5b6d
npm notice integrity:     sha512-0kRvMv6sI+dGw[...]EnNDIZkXcI+AQ==
npm notice total files:   9                                       
npm notice 
focus-trap-2.4.3.tgz
```

**Pull request merged**